linera-views/src/key_value_store_view.rs

```diff
@@ -467,6 +557,88 @@ where
         Ok(result)
     }
 
+    /// Applies the given batch of `crate::common::WriteOperation`.
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::create_memory_context;
+    /// # use linera_views::key_value_store_view::KeyValueStoreView;
+    /// # use linera_views::batch::Batch;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_memory_context();
+    ///   let mut view = KeyValueStoreView::load(context).await.unwrap();
+    ///   view.insert(vec![0,1], vec![34]).await.unwrap();
+    ///   view.insert(vec![3,4], vec![42]).await.unwrap();
+    ///   let mut batch = Batch::new();
+    ///   batch.delete_key_prefix(vec![0]);
+    ///   view.write_batch(batch).await.unwrap();
+    ///   let key_values = view.find_key_values_by_prefix(&[0]).await.unwrap();
+    ///   assert_eq!(key_values, vec![]);
+    /// # })
+    /// ```
+    pub async fn write_batch(&mut self, batch: Batch) -> Result<(), ViewError> {
+        *self.hash.get_mut() = None;
+        let max_key_size = self.max_key_size();
+        for operation in batch.operations {
+            match operation {
+                WriteOperation::Delete { key } => {
+                    ensure!(key.len() <= max_key_size, ViewError::KeyTooLong);
+                    if let Some(size) = self.sizes.get(&key).await? {
+                        self.total_size.0 -= size.0;
+                        self.total_size.1 -= size.1;
```

Comment:
nit: We could define `SubAssign` on the struct I mentioned above.

Reply:
Ok, but I named them `add_assign` and `sub_assign`.

Reply:
I think he meant the `AddAssign` and `SubAssign` traits, so you can write `-=` here.

Reply:
yes in std::ops